sklearn/metrics/_pairwise_distances_reduction/_pairwise_distances.pyx.tp

```diff
@@ -0,0 +1,183 @@
+cimport numpy as cnp
+from cython cimport final
+from cython.parallel cimport prange
+from ...utils._typedefs cimport ITYPE_t, DTYPE_t
+
+import numpy as np
+
+from scipy.sparse import issparse
+from sklearn import get_config
+from ...utils import check_array, _in_unstable_openblas_configuration
+from ...utils._openmp_helpers import _openmp_effective_n_threads
+from ...utils.fixes import threadpool_limits, sp_version, parse_version
+from ...utils._typedefs import DTYPE
+
+cnp.import_array()
+
+
+def _precompute_metric_params(X, Y, metric=None, **kwds):
+    """Precompute data-derived metric parameters if not provided."""
+    if metric == "seuclidean" and "V" not in kwds:
+        # There is a bug in scipy < 1.5 that will cause a crash if
+        # X.dtype != np.double (float64). See PR #15730
+        dtype = np.float64 if sp_version < parse_version("1.5") else None
+        if X is Y:
+            V = np.var(X, axis=0, ddof=1, dtype=dtype)
+        else:
+            raise ValueError(
+                "The 'V' parameter is required for the seuclidean metric "
+                "when Y is passed."
+            )
+        return {"V": V}
+    if metric == "mahalanobis" and "VI" not in kwds:
+        if X is Y:
+            VI = np.linalg.inv(np.cov(X.T)).T
+        else:
+            raise ValueError(
+                "The 'VI' parameter is required for the mahalanobis metric "
+                "when Y is passed."
+            )
+        return {"VI": VI}
+    return {}
+
+{{for name_suffix, INPUT_DTYPE in (('64', 'DTYPE'),('32', 'np.float32'))}}
+
+from ._datasets_pair cimport DatasetsPair{{name_suffix}}
+
+
+cdef class PairwiseDistances{{name_suffix}}:
+    """float{{name_suffix}} implementation of PairwiseDistances."""
+
+    @classmethod
+    def compute(
+        cls,
+        X,
+        Y,
+        str metric="euclidean",
+        chunk_size=None,
```

Comment:
Let's remove this for now.
```suggestion
```
I think we can reintroduce it for the Euclidean specialisation.